wicket/README.md

```diff
@@ -99,20 +99,38 @@ return actions when they need to be redrawn. However, the `Runner` also doesn't
 know when a screen animation is ongoing, and so it forwards all ticks to the
 `Screen` which returns an `Action::Redraw` if a redraw is necessary.
 
-# Testing wicket as a login shell
+# Testing wicket as a captive shell
 
-Wicket is meant to be used as a login shell. To test the login shell on a local Unix machine:
+Wicket is meant to be used as a captive shell over ssh. To test the captive shell support on a local Unix machine:
 
 1. Make the `wicket` available globally, at e.g. `/usr/local/bin/wicket`:
     * If your build directory is globally readable, create a symlink to `wicket` in a well-known location. From omicron's root, run: `sudo ln -s $(readlink -f target/debug/wicket) /usr/local/bin/wicket`
     * If it isn't globally accessible, run `sudo cp target/debug/wicket /usr/local/bin`. (You'll have to copy `wicket` each time you build it.)
 2. Add a new user to test against, for example `wicket-test`:
     1. Add a group for the new user: `groupadd wicket-test`.
-    2. Add the user: `sudo useradd -m -g wicket-test -s /usr/local/bin/wicket wicket-test`
-
-At this point, you can use `sudo -u wicket-test -i` (Linux) or `pfexec su - wicket-test` (illumos) to test wicket as a login shell.
-
-* A plain `sudo -u wicket-test -i` will show the TUI.
-* `sudo -u wicket-test -i upload ...` will let you upload an artifact over stdin.
-
-If you'd like to test connections over ssh, add your ssh key to the new user's `.ssh/authorized_keys`, then run `ssh wicket-test@localhost [upload ...]`.
+    2. Add the user: `sudo useradd -m -g wicket-test`
+3. Set up ssh authentication for this user, using either password-based auth or public-key based authentication using `.ssh/authorized_keys`.
+    * To configure ssh keys, you'll need to first log in as the `wicket-test` user. To do so, run `sudo -u wicket-test -i` (Linux) or `pfexec su - wicket-test` (illumos).
+    * If using `authorized_keys`, be sure to set up the correct permissions for `~/.ssh` and its contents. As the `wicket-test` user, run `chmod go-rwx -R ~/.ssh`.
+4. Test that you can log in as the user: run `ssh wicket-test@localhost`.
+    * To debug issues related to logging in, e.g. `~/.ssh` permissions issues, check the sshd authentication log.
+    * On Linux, the authentication log is typically at `/var/log/auth.log`.
+    * On illumos, the authentication log is at `/var/log/authlog`. If it is empty, logging needs to be enabled. (If you're an Oxide employee, see [this issue](https://github.com/oxidecomputer/helios-engvm/issues/18) for how to enable logging.)
+5. Add this to the end of `/etc/ssh/sshd_config`:
+    ```
+    Match User wicket-test
+        ForceCommand /usr/local/bin/wicket
```

Comment:
This is going to be different for illumos most likely, as there's no `/usr/local` on my helios box at least. Maybe just make it more open ended?

Reply:
I think step 1 kinda does make it open-ended; you could also create `/usr/local/bin` if you want.

Reply:
Yeah mostly I wanted to pick a concrete example while leaving it open-ended. I've reworded things a bit to make this clearer.